File: downloads-that-dont-fail-automating-file-verification/javascript/fileVerification.js
const fs = require("fs");
const path = require("path");

// Define the expected file name and path
const expectedFileName = "sample-file.txt";
const downloadDirectory = "/path/to/downloads/";
const filePath = path.join(downloadDirectory, expectedFileName);

// Check if the file exists and has the correct extension
if (fs.existsSync(filePath) && filePath.endsWith(".txt")) {
  console.log(`File name and type verification passed: ${expectedFileName}`);
} else {
  throw new Error(`File name or type verification failed: ${filePath}`);
}

// Read the downloaded file and check its content
const fileContent = fs.readFileSync(filePath, "utf-8");

if (fileContent.includes("testing download")) {
  console.log("Text content verification passed");
} else {
  throw new Error("Text content verification failed");
}

// Get the size of the downloaded file
const stats = fs.statSync(filePath);
const fileSizeInBytes = stats.size;

// Ensure the file size is greater than 0 bytes
if (fileSizeInBytes > 0) {
  console.log(`File size verification passed: ${fileSizeInBytes} bytes`);
} else {
  throw new Error("File size verification failed: File is empty or corrupted");
}

File: downloads-that-dont-fail-automating-file-verification/python/file_verification.py
import os

# Define the expected file name and path
expected_file_name = "sample-file.txt"
download_directory = "/path/to/downloads"
file_path = os.path.join(download_directory, expected_file_name)

# Check if the file exists and has the correct name and extension
if os.path.exists(file_path) and file_path.endswith('.txt'):
    print(f"File name and type verification passed: {expected_file_name}")
else:
    raise Exception(f"File name or type verification failed: {file_path}")

# Open the downloaded file and check its contents
with open(file_path, 'r') as file:
    content = file.read()

if "expected text" in content:
    print("Text content verification passed")
else:
    raise Exception("Text content verification failed")

# Check the size of the downloaded file
file_size = os.path.getsize(file_path)

# Ensure the file size is greater than 0 bytes
if file_size > 0:
    print(f"File size verification passed: {file_size} bytes")
else:
    raise Exception("File size verification failed: File is empty or corrupted")
